$MirOS$

	• housekeeping for MPFR’s patch system
	• inlining MPFR 3.1’s __gmp_const patch

--- mpfr.h.orig	Thu May 29 18:52:22 2014
+++ mpfr.h	Thu May 29 19:23:19 2014
@@ -1,5 +1,8 @@
 /* mpfr.h -- Include file for mpfr.
 
+Copyright © 2014
+	Thorsten “mirabilos” Glaser <tg@mirbsd.org>
+
 Copyright 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009 Free Software Foundation, Inc.
 Contributed by the Arenaire and Cacao projects, INRIA.
 
@@ -27,7 +30,7 @@ MA 02110-1301, USA. */
 #define MPFR_VERSION_MAJOR 2
 #define MPFR_VERSION_MINOR 4
 #define MPFR_VERSION_PATCHLEVEL 2
-#define MPFR_VERSION_STRING "2.4.2-p3"
+#define MPFR_VERSION_STRING "2.4.2-p3+rh643657+__gmp_const"
 
 /* Macros dealing with MPFR VERSION */
 #define MPFR_VERSION_NUM(a,b,c) (((a) << 16L) | ((b) << 8) | (c))
@@ -37,6 +40,18 @@ MPFR_VERSION_NUM(MPFR_VERSION_MAJOR,MPFR
 /* Check if GMP is included, and try to include it (Works with local GMP) */
 #ifndef __GMP_H__
 # include <gmp.h>
+#endif
+
+/* GMP's internal __gmp_const macro has been removed on 2012-03-04:
+     http://gmplib.org:8000/gmp/rev/d287cfaf6732
+   const is standard and now assumed to be available. If the __gmp_const
+   definition is no longer present in GMP, this probably means that GMP
+   assumes that const is available; thus let's define it to const.
+   Note: this is a temporary fix that can be backported to previous MPFR
+   versions. In the future, __gmp_const should be replaced by const like
+   in GMP. */
+#ifndef __gmp_const
+# define __gmp_const const
 #endif
 
 /* Check if stdio.h is included or if the user wants FILE */
